### PR TITLE
Miscellaneous cleanups

### PR DESCRIPTION
**Pull request by jsquyres**
Small cleanups found while working on #10223, #10224, and #10245.  See individual commit messages, the most important of which is likely the MacOS `/usr/bin/python` fix.

### CODE REVIEW

**Comment by bwbarrett** [comment]
bot:aws:retest

File: configure.ac [hunk 1]
```diff
@@ -128,19 +128,44 @@ AC_DEFUN([OMPI_CHECK_DIR_FOR_SPACES],[
...
+        #   "realpath" in the base OS.
+        # - MacOS Monterrey / 12.3 does support "readlink -f", but
+        #   "realpath" is still not in the base OS.
+        #
+        # Instead of trying to figure out if "readlink -f" is
+        # supported, use Python: MacOS will have either
+        # /usr/bin/python (e.g., MacOS Catalina / 10.15) or
+        # /usr/bin/python3 (e.g., MacOS Monterrey / 12.3), both of
+        # which have os.path.realpath.
+        AS_IF([test -x /usr/bin/python],
```
Comment (bwbarrett) [inline configure.ac:150]: Ralph and I worked through this in OpenPMIx and came up with something way simpler: https://github.com/openpmix/openpmix/blob/7bd5c452b302f656a8dd5c8e2a98b92a545143b0/configure.ac#L136

**Review by bwbarrett — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (jsquyres) [reply configure.ac:150]: There is no `python` executable in MacOS 12.3.1 Monterrey:

```
$ git clone git@github.com:openpmix/openpmix.git
...snipped...
$ cd openpmix
$ ./autogen.pl
...snipped...
$ ./configure
: builddir: /Users/jsquyres/git/openpmix
: srcdir: /Users/jsquyres/git/openpmix

============================================================================
== Configuring PMIx
============================================================================
checking build system type... x86_64-apple-darwin21.4.0
...snipped...
checking directory of source tree... .
./configure: line 6702: python: command not found
checking directory of absolute source tree... 
./configure: line 6734: python: command not found
checking directory of build tree... 
checking directory of prefix... /Users/jsquyres/bogus
./configure: line 6784: python: command not found
checking directory of absolute prefix... 
```

Comment (jsquyres) [reply configure.ac]: @bwbarrett and I talked out of band: this commit now removes all the path-canonicalizing stuff (technically, it wasn't needed in the first place).  That avoids the issue altogether.

File: ompi/mpi/java/c/mpi_Info.c [hunk 2]
```diff
@@ -45,17 +45,20 @@ JNIEXPORT jlong JNICALL Java_mpi_Info_getNull(JNIEnv *env, jclass clazz)
...
+// At least some versions of jni.h have a global named "jvalue", and
+// we get a compiler warning if we have a parameter or variable of the
+// same name.  So use "ljvalue" instead.
 JNIEXPORT void JNICALL Java_mpi_Info_set(
-        JNIEnv *env, jobject jthis, jlong handle, jstring jkey, jstring jvalue)
+        JNIEnv *env, jobject jthis, jlong handle, jstring jkey, jstring ljvalue)
 {
-    const char *key   = (*env)->GetStringUTFChars(env, jkey,   NULL),
-               *value = (*env)->GetStringUTFChars(env, jvalue, NULL);
+    const char *key   = (*env)->GetStringUTFChars(env, jkey,    NULL),
```
Comment (bwbarrett) [inline ompi/mpi/java/c/mpi_Info.c:54]: why the extra spaces before NULL?

File: ompi/mpi/java/c/mpi_Info.c [hunk 3]
```diff
@@ -45,17 +45,20 @@ JNIEXPORT jlong JNICALL Java_mpi_Info_getNull(JNIEnv *env, jclass clazz)
...
+               *value = (*env)->GetStringUTFChars(env, ljvalue, NULL);
 
     int rc = MPI_Info_set((MPI_Info)handle, (char*)key, (char*)value);
     ompi_java_exceptionCheck(env, rc);
 
-    (*env)->ReleaseStringUTFChars(env, jkey,   key);
-    (*env)->ReleaseStringUTFChars(env, jvalue, value);
+    (*env)->ReleaseStringUTFChars(env, jkey,    key);
```
Comment (bwbarrett) [inline ompi/mpi/java/c/mpi_Info.c:60]: Why the extra spaces before key?

File: config/ompi_setup_prrte.m4 [hunk 4]
```diff
@@ -42,7 +42,7 @@ AC_DEFUN([OMPI_SETUP_PRRTE],[
     OPAL_3RDPARTY_WITH([prrte], [prrte], [package_prrte], [1])
 
     AC_ARG_WITH([prrte-bindir],
-       [AC_HELP_STRING([--with-prrte-bindir=DIR],
+       [AS_HELP_STRING([--with-prrte-bindir=DIR],
```
Comment (bwbarrett) [inline config/ompi_setup_prrte.m4:45]: shame on me :(

**Review by bwbarrett — APPROVED** [review]

File: ompi/mpi/java/c/mpi_Info.c (hunk 2 above)
Comment (jsquyres) [reply ompi/mpi/java/c/mpi_Info.c:54]: I just maintained the same spacing that was there from the original author: that the NULL's on the 2 successive lines were aligned.

I don't know if this style is uniform throughout the Java bindings, but it does occur in at least a few places that I spot checked.  Want me to remove them here?  I don't have strong feelings, either way.

Comment (bwbarrett) [reply ompi/mpi/java/c/mpi_Info.c:54]: nah, don't worry about it.  Although in the diff, it looks like you added a space.

**Pull request merged**